allennlp/modules/token_embedders/embedding.py

```diff
@@ -181,18 +181,24 @@ def extend_vocab(self,  # pylint: disable=arguments-differ
         """
         # Caveat: For allennlp v0.8.1 and below, we weren't storing vocab_namespace as an attribute,
         # knowing which is necessary at time of embedding vocab extension. So old archive models are
-        # currently unextendable unless the user used default vocab_namespace 'tokens' for it.
+        # currently unextendable.
 
         vocab_namespace = vocab_namespace or self._vocab_namespace
         if not vocab_namespace:
-            vocab_namespace = "tokens"
-            logging.warning("No vocab_namespace provided to Embedder.extend_vocab. Defaulting to 'tokens'.")
+            # It's not safe to default to 'tokens' when we aren't sure that 'tokens'
+            # need to be extended. (Without this, several tests fail.)
```

Comment:
No need to reference failing tests in comments in the code - just give the justification (that it's not safe to default to "tokens").